lib/iris/fileformats/_ff.py

```diff
@@ -717,76 +717,85 @@ def _extract_field(self):
-                stash_entry = STASH_TRANS.get(stash, None)
-                if stash_entry is None:
-                    subgrid = None
-                else:
-                    subgrid = stash_entry.grid_code
-                    if subgrid not in HANDLED_GRIDS:
-                        warnings.warn('The stash code {} is on a grid {} '
-                                      'which has not been explicitly handled '
-                                      'by the fieldsfile loader. Assuming the '
-                                      'data is on a P grid.'.format(stash,
-                                                                    subgrid))
-
-                field.x, field.y = grid.vectors(subgrid)
-
-                # Use the per-file grid if no per-field metadata is available.
-                no_x = field.bzx in (0, field.bmdi) and field.x is None
-                no_y = field.bzy in (0, field.bmdi) and field.y is None
-                if no_x and no_y:
-                    if subgrid is None:
-                        msg = ('The STASH code {0} was not found in the '
-                               'STASH to grid type mapping. Picking the P '
-                               'position as the cell type'.format(stash))
-                        warnings.warn(msg)
-                    field.bzx, field.bdx = grid.regular_x(subgrid)
-                    field.bzy, field.bdy = grid.regular_y(subgrid)
-                    field.bplat = grid.pole_lat
-                    field.bplon = grid.pole_lon
-                elif no_x or no_y:
-                    warnings.warn(
-                        'Partially missing X or Y coordinate values.')
-
-                # Check for LBC fields.
-                is_boundary_packed = self._ff_header.dataset_type == 5
-                if is_boundary_packed:
-                    # Apply adjustments specific to LBC data.
-                    self._adjust_field_for_lbc(field)
-
-                # Calculate start address of the associated PP header data.
-                data_offset = field.lbegin * self._word_depth
-                # Determine PP field payload depth and type.
-                data_depth, data_type = self._payload(field)
-
-                # Produce (yield) output fields.
-                if is_boundary_packed:
-                    fields = self._fields_over_all_levels(field)
-                else:
-                    fields = [field]
-                for result_field in fields:
-                    # Add a field data element.
-                    if self._read_data:
-                        # Read the actual bytes. This can then be converted to
-                        # a numpy array at a higher level.
-                        ff_file_seek(data_offset, os.SEEK_SET)
-                        result_field._data = pp.LoadedArrayBytes(
-                            ff_file.read(data_depth), data_type)
+                # (Some Fields File fields are UM specific scratch spaces, with
```

Comment:
Shouldn't state here what the underlying problem is, as that is the PP module's business.
(see other comment)...
We should just say that the field creation failed here.